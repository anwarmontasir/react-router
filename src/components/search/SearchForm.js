import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './SearchForm.css';

export default class SearchForm extends Component {

  static propTypes = {
    searchTerm: PropTypes.string,
    onSearch: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      current: props.searchTerm || ''
    };
  }

  componentWillReceiveProps({ searchTerm }) {
    if(searchTerm !== this.state.current) {
      this.setState({ current: searchTerm || '' });
    }
  }

  handleSubmit = event => {
    event.preventDefault();
    const { current } = this.state;
    if(!current) return;
    this.props.onSearch(current);
  }

  render() {
    const { current } = this.state;

    return (
      <form className={styles.search}
        onSubmit={this.handleSubmit}
      >
        <fieldset>
          <label>
            Search for Movies: <input value={current} onChange={({ target }) => this.setState({ current: target.value })} name="search" placeholder="enter movie search"/>
          </label>
          <label>
            &nbsp;<button>Search</button>
          </label>
        </fieldset>
      </form>
    );
  }
}