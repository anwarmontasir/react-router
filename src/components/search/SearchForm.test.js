import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SearchForm from './SearchForm';

describe('Search Form', () => {
  
  it('renders as design', () => {
    const wrapper = shallow(<SearchForm onSearch={() => {}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Calls onSearch with criteria entered', () => {
    let filter;
    const handleSearch = _filter => filter = _filter;

    const wrapper = mount(<SearchForm onSearch={handleSearch}/>);

    const search = 'Star Wars';

    // input and click
    const input = wrapper.find('input');
    input.instance().value = search;
    const form = wrapper.find('form').instance();
    const button = wrapper.find('button');
    button.simulate('submit', { target: form });

    // did the handleSearch get called with the simulated input?
    expect(filter).toBe(search);
  });
});