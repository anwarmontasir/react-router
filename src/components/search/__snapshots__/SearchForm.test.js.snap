// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders as design 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <label>
      Search for Movies: 
      <input
        name="search"
        onChange={[Function]}
        placeholder="enter movie search"
        value=""
      />
    </label>
    <label>
      Â 
      <button>
        Search
      </button>
    </label>
  </fieldset>
</form>
`;
